tests/test-cases.js

```diff
@@ -22,6 +22,61 @@ try {
 	XHR_ENABLED = true;
 } catch (e) {} // safari, ie
 
+var __substring = String.prototype.substring;
```

Comment:
Why is this all stuff neeeded?

Reply:
it allows the unit test to fail if substring is called with bad inputs, which is the first issue being addressed.